app/sync.js

```diff
@@ -116,42 +118,52 @@ const appStoreChangeCallback = function (diffs) {
         const entryJS = entry.toJS()
         entryJS.objectId = entryJS.objectId || syncUtil.newObjectId(statePath)
 
-        sendSyncRecords(backgroundSender, action,
-          [type === 'sites' ? syncUtil.createSiteData(entryJS) : syncUtil.createSiteSettingsData(statePath[1], entryJS)])
+        pendingRecords = pendingRecords.concat(
+          sendSyncRecords(backgroundSender, action,
+            [type === 'sites' ? syncUtil.createSiteData(entryJS) : syncUtil.createSiteSettingsData(statePath[1], entryJS)])
+        )
       }
     }
   })
+  if (pendingRecords.length > 0) {
+    appActions.pendSyncRecords(pendingRecords)
+  }
 }
 
 /**
  * Sends sync records of the same category to the sync server.
  * @param {event.sender} sender
  * @param {number} action
  * @param {Array.<{name: string, value: Object}>} data
+ * @returns {Array.<object>} records which were sent
  */
 const sendSyncRecords = (sender, action, data) => {
   if (!deviceId) {
     throw new Error('Cannot build a sync record because deviceId is not set')
   }
   if (!data || !data.length || !data[0]) {
-    return
+    return []
   }
   const category = CATEGORY_MAP[data[0].name]
   if (!category ||
     (category.settingName && !getSetting(settings[category.settingName]))) {
-    return
+    return []
   }
-  sender.send(syncMessages.SEND_SYNC_RECORDS, category.categoryName, data.map((item) => {
-    if (!item || !item.name || !item.value) {
+  const records = []
+  data.forEach(item => {
+    if (!item || !item.name || !item.value || !item.objectId) {
       return
     }
-    return {
+    records.push({
       action,
       deviceId,
       objectId: item.objectId,
       [item.name]: item.value
-    }
-  }))
+    })
+  })
+  log(`Sending ${records.length} sync records`)
+  sender.send(syncMessages.SEND_SYNC_RECORDS, category.categoryName, records)
```

Comment:
why not just call `appActions.pendSyncRecords` in here instead of returning the records and calling it every time that sendSyncRecords is invoked?